content/concepts/connections.md

```diff
@@ -0,0 +1,159 @@
+and does not enforce the implementation of a specific transport protocol. 
+Though the support of modern transport protocols is a primary focus for 
+libp2p, implementers may choose to support multiple types of transport. Learn 
+more about transport protocols in libp2p on the 
+[transport guide]((/concepts/transports/)). 
+
+Let's first outline key terminology to help us understand connections in libp2p.
+
+## Core libp2p components
+
+- **Peer**: a participant in a libp2p network.
+- **Peer-to-Peer (P2P)**: a distributed network in which workloads are 
+  shared between *Peers*.
+- **Connection**: a network layer connection between two peers in a libp2p 
+  network.
+- **Initiator**: a peer that initiates a connection through a dial request.
+- **Responder**: a peer that listens to dial requests.
+- **Transport**: a mechanism of communication that creates a *Connection*, 
+  e.g., TCP, QUIC, WebRTC, etc.
+- **Stream**: an abstraction on top of a *Connection* representing a bidirectional 
+  communication channel between two *Peers*.
+- **Multiplexer**: a mechanism that manages multiple streams and ensures parallel 
+  messages reach the correct *Stream*, based on stream identification.
+- **Secure channel**: establishes a secure, encrypted, and authenticated channel 
+  over a *Connection*.
+- **Upgrader**: a mechanism that takes a *Connection* returned by a *Transport*, 
+  and performs protocol negotiation to set up a secure, multiplexed channel on 
+  top of which *Streams* can be opened.
+
+## Opening a libp2p connection
+
+A libp2p connection only requires two core operations: dialing and listening. 
+Both operations exist as interfaces in libp2p, where a transport protocol is used 
+to expose the dialing and listening interfaces. The listening interface allows a 
+peer to listen to connection requests from others, while the dialing interface 
+allows a peer to send connection requests to a listening peer. Most of the connection 
+initation process is abstracted through libp2p's interfaces.
+
+### How do peers prevent multiple connections from being established? 
+
+An interface called a [switch](/concepts/stream-multiplexing/#switch/swarm) 
+(and sometimes swarm) implements a basic dial queue that manages an application's 
+dialing and listening state. This prevents multiple, 
+simultaneous dial requests to the same peer. Peers will generally dial other
+peers using the switch interface, which initiates a stream to read from and write 
+to. The switch will find the appropriate transport protocol to open a connection. 
+
+### How do peers know what transports other peers support?
+
+A data store known as a Peer Store in libp2p holds an updated data registry of 
+all known peers and their Peer Info. Other peers can dial the Peer Store, listen 
+for updates, and learn about any peer within the network. More information about peers
+is available on the [peers guide](/concepts/peers).
```

Comment:
As a data point, rust-libp2p does not have a _Peer Store_. May this be too detailed, i.e. to specific to a concrete implementation?